web3/contract.py

```diff
@@ -645,6 +665,103 @@ def _encode_constructor_data(cls, args=None, kwargs=None):
         return deploy_data
 
 
+class ContractConstructor:
+    """
+    Class for contract constructor API.
+    """
+    def __init__(self, web3, address, abi, bytecode, args, kwargs):
+        self.web3 = web3
+        self.address = address
+        self.abi = abi
+        self.bytecode = bytecode
+        self.data_in_transaction = self._encode_data_in_transaction(args, kwargs)
+
+    @combomethod
+    def _encode_data_in_transaction(self, args=None, kwargs=None):
+        constructor_abi = get_constructor_abi(self.abi)
+
+        if constructor_abi:
+            if args is None:
+                args = tuple()
+            if kwargs is None:
+                kwargs = {}
+
+            arguments = merge_args_and_kwargs(constructor_abi, args, kwargs)
+            data = add_0x_prefix(
+                encode_abi(self.web3, constructor_abi, arguments, data=self.bytecode)
+            )
+        else:
+            data = to_hex(self.bytecode)
+
+        return data
+
+    @combomethod
+    def estimateGas(self, transaction=None):
+        if transaction is None:
+            estimate_gas_transaction = {}
+        else:
+            estimate_gas_transaction = dict(**transaction)
+            self.check_forbidden_keys_in_transaction(estimate_gas_transaction,
+                                                     ["data", "to"])
+
+        if self.address:
+            estimate_gas_transaction.setdefault('to', self.address)
```

Comment:
These two lines don't make sense to me in the context of a constructor, which should always have a blank `to` field.